web_monitoring/internetarchive_alt.py

```diff
@@ -0,0 +1,162 @@
+"""
+This module provides a Python API for accessing versions (timestamped captures)
+of a URL. There are existing open-source Python packages for the Internet
+Archive API (the best-established one seems to be
+https://internetarchive.readthedocs.io/en/latest/) but none that expose the
+list of versions of a URL.
+
+References used in writing this module:
+* https://ws-dl.blogspot.fr/2013/07/2013-07-15-wayback-machine-upgrades.html
+
+Other potentially useful links:
+* https://blog.archive.org/developers/
+* https://archive.readme.io/docs/memento
+"""
+
+from datetime import datetime
+import re
+import requests
+from web_monitoring import utils
+
+class WebMonitoringException(Exception):
+    # All exceptions raised directly by this package inherit from this.
+    ...
+
+
+class UnexpectedResponseFormat(WebMonitoringException):
+    ...
+
+
+TIMEMAP_URL_TEMPLATE = 'http://web.archive.org/web/timemap/link/{}'
+DATE_FMT = '%a, %d %b %Y %H:%M:%S %Z'
+DATE_URL_FMT = '%Y%m%d%H%M%S'
+URL_CHUNK_PATTERN = re.compile('\<(.*)\>')
+DATETIME_CHUNK_PATTERN = re.compile(' datetime="(.*)",')
+
+
+def list_versions(url):
+    """
+    Yield (version_datetime, version_uri) for all versions of a url.
+
+    Parameters
+    ----------
+    url : string
+
+    Examples
+    --------
+    Grab the datetime and URL of the version nasa.gov snapshot.
+    >>> pairs = list_versions('nasa.gov')
+    >>> dt, url = next(pairs)
+    >>> dt
+    datetime.datetime(1996, 12, 31, 23, 58, 47)
+    >>> url
+    'http://web.archive.org/web/19961231235847/http://www.nasa.gov:80/'
+
+    Loop through all the snapshots.
+    >>> for dt, url in list_versions('nasa.gov'):
+    ...     # do something
+
+    References
+    ----------
+    * https://ws-dl.blogspot.fr/2013/07/2013-07-15-wayback-machine-upgrades.html
+    """
+    # Request a list of the 'mementos' (what we call 'versions') for a url.
+    # It may be paginated. If so, the final line in the repsonse is a link to
+    # the next page.
+    
+
+    first_page_url = TIMEMAP_URL_TEMPLATE.format(url)
+    res = requests.get(first_page_url)
+    lines = res.iter_lines()
+
+    try:
+        # The first three lines contain no information we need.
+        for _ in range(3):
+            next(lines)
+    except:
```

Comment:
It is best to be as specific as possible when catching errors. We expect `StopIteration` here. If we got some _other_ kind of error, we would want that error to continue to propagate so it could be debugged.